docs/webpack.md

```diff
@@ -256,6 +256,47 @@ const { environment } = require('@rails/webpacker')
 environment.resolvedModules.append('vendor', 'vendor')
 ```
 
+### Add SplitChunks (Webpack V4)
+Originally, chunks (and modules imported inside them) were connected by a parent-child relationship in the internal webpack graph. The CommonsChunkPlugin was used to avoid duplicated dependencies across them, but further optimizations were not possible
+
+Since webpack v4, the CommonsChunkPlugin was removed in favor of optimization.splitChunks.
+
+For the full configuration options of SplitChunks, see the [Webpack documentation](https://webpack.js.org/plugins/split-chunks-plugin/).
+
+```js
+// config/webpack/environment.js
+const WebpackAssetsManifest = require('webpack-assets-manifest');
+
+const splitChunks = {
+  optimization: {
+    splitChunks: {
+      chunks: 'all'
+    },
+  },
+};
+
+environment.config.merge(splitChunks);
+
+// Should override the existing manifest plugin
+environment.plugins.insert(
+  'Manifest',
+  new WebpackAssetsManifest({
+    entrypoints: true, // default in rails is false
+    writeToDisk: true, // rails defaults copied from webpacker
+    publicPath: true // rails defaults copied from webpacker
+  })
+)
+```
+
+The `javascript_pack_tag` and `stylesheet_pack_tag` doesn't know (by default) whether they should use the entrypoints or that it should use the normal way of adding a pack. To make sure the `*_pack_tag` works with `SplitChunks`, you should use the following syntax;
+
+```erb
+javascript_pack_tag "your-entrypoint-javascript-file.js", split_chunks: true
+stylesheet_pack_tag "your-entrypoint-stylesheet-file.css", split_chunks: true
+```
+
+For the old configuration with the CommonsChunkPlugin see below
```

Comment:
You might want to remove this since you can't use CommonsChunkPlugin with webpack 4+

Reply:
Wouldn’t be a depreciation notice be a better option here? 

The publicly available version of webpacker is still the version that is not compatible with webpack 4. 

I would suggest to add a deprecation notice until the webpack 4 version of webpacker is out of beta. As Github is the only source for documentation. 

But; I’m not the expert on maintaining large OS projects like these, so I’m fine with both options.